### PR TITLE
Resolve some existing compiler warnings

### PR DESCRIPTION
**Pull request by kent-mcleod**

### CODE REVIEW

**Review by ikuz — APPROVED** [review]

File: libsel4debug/src/backtrace.c [hunk 1]
```diff
@@ -4,12 +4,18 @@
  * SPDX-License-Identifier: BSD-2-Clause
  */
 #include <autoconf.h>
+#include <sel4/types.h>
+
+/* Don't instrument seL4_GetIPCBuffer so we don't recurse.
```
Comment (axel-h) [inline libsel4debug/src/backtrace.c:9]: Can you give a few more details where/why the recrusion happens?

File: libsel4debug/src/backtrace.c [hunk 2]
```diff
@@ -4,12 +4,18 @@
...
+ * This definition has to come before the definition that is included
+ * for the attribute to have effect.
+ */
+LIBSEL4_INLINE_FUNC seL4_IPCBuffer *seL4_GetIPCBuffer(void) __attribute__((no_instrument_function));
```
Comment (axel-h) [inline libsel4debug/src/backtrace.c:13]: Why can't we drop the prototype here completely and just add the attribute in kernel/libsel4/include/sel4/functions.h where the implementation is?

Comment (kent-mcleod) [reply libsel4debug/src/backtrace.c:13]: It's specific to the file.  The functions __cyg_profile_func_enter and __cyg_profile_func_exit call seL4_GetIPCBuffer and need to ensure that the compiler hasn't instrumented it.

File: libsel4debug/src/backtrace.c (hunk 1 above)
Comment (kent-mcleod) [reply libsel4debug/src/backtrace.c:9]: I updated the comment.

**Comment by axel-h** [comment]
I was wondering if changing kernel/libsel4/include/sel4/functions.h to have 
```
LIBSEL4_INLINE_FUNC void seL4_SetIPCBuffer(seL4_IPCBuffer *ipc_buffer) LIBSEL4_ATTRIBUTES_SetIPCBuffer
{
    __sel4_ipc_buffer = ipc_buffer;
    return;
}
```

and then just setting 
```
#define LIBSEL4_ATTRIBUTES_SetIPCBuffer __attribute__((no_instrument_function))
```
might be a better approach to avoid redundancy. But I'm not really sure it this is better to understand without a few comments why this is needed.

**Comment by kent-mcleod** [comment]
I don't think it's worth complicating the original definition just to make it easier for this file to do something non-standard.

**Review by axel-h — APPROVED** [review]

**Pull request merged**